src/rpdk/core/contract/resource_client.py

```diff
@@ -290,6 +290,18 @@ def is_primary_identifier_equal(
             for primary_identifier in primary_identifier_path
         )
 
+    @staticmethod
+    def assert_write_only_property_does_not_exist(
+        resource_model, write_only_properties
+    ):
+        if write_only_properties:
+            assert not all(
```

Comment:
the way all() works is all evaluations have to satisfy. that means one writeOnlyProperty exists but the rest don't this would evaluate to false and pass the assertion. I think you want to use any() here and if any properties are found fail.